File: src/main/Services/AIExternalService.cs
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using Microsoft.Extensions.Configuration;
using Microsoft.AspNetCore.Mvc;

using Lernassistent.src.main.Controller;

namespace Lernassistent.src.main.services
{
    public class AIExternalService
    {
        private readonly IHttpClientFactory _httpClientFactory;
        private readonly IConfiguration _configuration;

        public AIExternalService(IHttpClientFactory httpClientFactory, IConfiguration configuration)
        {
            _httpClientFactory = httpClientFactory;
            _configuration = configuration;
        }

        public async Task<IActionResult> CallExternalAI(AIRequest request, string extractedText)
        {
            try
            {
                var httpClient = _httpClientFactory.CreateClient();
                var apiKey = _configuration["ExternalAI:ApiKey"];
                httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);

                var aiRequest = new
                {
                    text = extractedText,
                    userId = request.UserId
                };
                var payload = new StringContent(JsonSerializer.Serialize(aiRequest), Encoding.UTF8, "application/json");

                var response = await httpClient.PostAsync($"{_configuration["ExternalAI:BaseUrl"]}/api/ai", payload);

                if (response.IsSuccessStatusCode)
                {
                    var responseContent = await response.Content.ReadAsStringAsync();
                    return new OkObjectResult(new { Message = "Request routed to external AI", Result = responseContent });
                }

                return new BadRequestObjectResult("External AI request failed");
            }
            catch (Exception ex)
            {
                return new ObjectResult($"External AI call error: {ex.Message}") { StatusCode = 500 };
            }
        }
    }
}

File: src/main/App/Services/AiExternalService.py
import requests

class AIExternalService:
    def __init__(self, api_key, api_url):
        self.api_key = api_key
        self.api_url = api_url

    async def process_text(self, text: str):
        """
        Send a request to the GroqCloud API to process the given text.
        """
        headers = {
            "Content-Type": "application/json",
            "Authorization": f"Bearer {self.api_key}"
        }
        payload = {
            "model": "llama3-8b-8192",
            "messages": [{"role": "user", "content": text}]
        }

        try:
            response = requests.post(self.api_url, json=payload, headers=headers)
            if response.status_code == 200:
                response_data = response.json()
                return response_data['choices'][0]['message']['content']
            else:
                return f"Error: {response.status_code} - {response.text}"
        except Exception as ex:
            return f"Exception: {str(ex)}"
